### ISSUE TITLE
New versions missing

### ISSUE THREAD

**Issue by TeoColuccio** [bkw777/mainline#175]
As per the title, looking for new versions of the kernel from mainline, I find 6.0.9 as the latest stable version, while it should be 6.1.3 and at the same time the latest available rc is 6.1.0-rc5, while on https://kernel.ubuntu.com/~kernel-ppa/mainline/ latest is 6.2-rc2.
This is the output of `mainline --list`:
```
teo@Msi-Pulse:~$ mainline --list
mainline 1.0.18
Distribution: Ubuntu 22.10
Architecture: amd64
Running kernel: 6.1.0-060100rc5-generic
Fetching index from kernel.ubuntu.com...
OK
----------------------------------------------------------------------
Found installed: 5.19.0-21.21
Found installed: 5.19.0.26.23
Found installed: 6.1.0-060100rc5.202211132230
Found installed: 6.0.0-060000.202210022231
Found installed: 5.19.0-26.27
----------------------------------------------------------------------
----------------------------------------------------------------------

======================================================================
Available Kernels
======================================================================
6.1.0-rc5                        Running
6.1.0-rc4                        
6.1.0-rc3                        
6.1.0-rc1                        
6.0.9                            
6.0.8                            
6.0.7                            
6.0.6                            
6.0.5                            
6.0.4                            
6.0.3                            
6.0.1                            
6.0.0                            Installed
6.0.0-rc7                        
6.0.0-rc6                        
6.0.0-rc5                        
6.0.0-rc4                        
6.0.0-rc3                        
5.19.17                          
5.19.16                          
5.19.15                          
5.19.13                          
5.19.11                          
5.19.10                          
5.19.9                           
5.19.8                           
5.19.7                           
5.19.6                           
5.19.5                           
5.19.3                           
5.19.2                           
5.19.1                           
5.19.0.26.27                     Installed
5.19.0.26.23                     Installed
5.19.0.21.21                     Installed
5.19.0                           
5.19.0-rc8                       
5.19.0-rc7                       
5.19.0-rc6                       
5.19.0-rc5                       
5.19.0-rc4                       
5.19.0-rc3                       
5.18.19                          
5.18.17                          
5.18.15                          
5.18.14                          
5.18.13                          
5.18.12                          
5.18.11                          
5.18.10                          
5.18.8                           
5.18.6                           
5.18.4                           
5.18.3                           
5.18.2                           
5.18.0                           
5.18.0-rc7                       
5.18.0-rc6                       
5.18.0-rc5                       
5.18.0-rc1                       
5.17.15                          
5.17.14                          
5.17.13                          
5.17.11                          
5.17.10                          
5.17.9                           
5.17.8                           
5.17.7                           
5.17.6                           
5.17.5                           
5.17.4                           
5.17.3                           
5.17.2                           
5.17.1                           
5.17.0                           
5.17.0-rc8                       
5.17.0-rc7                       
5.17.0-rc6                       
5.17.0-rc5                       
5.17.0-rc4                       
5.17.0-rc3                       
5.17.0-rc2                       
5.16.20                          
5.16.19                          
5.16.18                          
5.16.17                          
5.16.16                          
5.16.15                          
5.16.14                          
5.16.13                          
5.16.12                          
5.16.11                          
5.16.10                          
5.16.5                           
5.16.0                           
5.16.0-rc8                       
5.16.0-rc7                       
5.16.0-rc6                       
5.16.0-rc5                       
5.16.0-rc4                       
5.16.0-rc3                       
5.16.0-rc2                       
5.16.0-rc1                       
5.15.86                          
5.15.85                          
5.15.84                          
5.15.83                          
5.15.82                          
5.15.81                          
5.15.79                          
5.15.78                          
5.15.77                          
5.15.76                          
5.15.75                          
5.15.74                          
5.15.73                          
5.15.72                          
5.15.71                          
5.15.70                          
5.15.69                          
5.15.68                          
5.15.65                          
5.15.60                          
5.15.55                          
5.15.50                          
5.15.45                          
5.15.35                          
5.15.30                          
5.15.25                          
5.15.20                          
5.15.15                          
5.15.10                          
5.15.5                           
5.15.0                           
5.15.0-rc7                       
5.15.0-rc6                       
5.15.0-rc5                       
5.15.0-rc3                       
5.15.0-rc2                       
5.14.21                          
5.14.20                          
5.14.19                          
5.14.18                          
5.14.17                          
5.14.16                          
5.14.15                          
5.14.14                          
5.14.13                          
5.14.12                          
5.14.10                          
5.14.5                           
5.14.0                           
5.14.0-rc7                       
5.14.0-rc6                       
5.14.0-rc5                       
5.14.0-rc4                       
5.14.0-rc3                       
5.14.0-rc2                       
5.13.19                          
5.13.18                          
5.13.17                          
5.13.16                          
5.13.15                          
5.13.14                          
5.13.13                          
5.13.12                          
5.13.11                          
5.13.9                           
5.13.5                           
5.13.0                           
5.13.0-rc7                       
5.13.0-rc6                       
5.13.0-rc5                       
5.13.0-rc4                       
5.13.0-rc3                       
5.13.0-rc2                       
5.13.0-rc1                       
5.12.19                          
5.12.18                          
5.12.17                          
5.12.16                          
5.12.15                          
5.12.14                          
5.12.13                          
5.12.12                          
5.12.11                          
5.12.10                          
5.12.5                           
5.12.0                           
5.12.0-rc8                       
5.12.0-rc7                       
5.12.0-rc6                       
5.12.0-rc5                       
5.12.0-rc4                       
5.12.0-rc3                       
5.12.0-rc2                       
5.12.0-rc1                       
5.12.0-rc1                       
5.11.20                          
5.11.19                          
5.11.18                          
5.11.17                          
5.11.16                          
5.11.15                          
5.11.14                          
5.11.13                          
5.11.10                          
5.11.5                           
5.11.0                           
5.11.0-rc7                       
5.11.0-rc6                       
5.11.0-rc5                       
5.11.0-rc4                       
5.11.0-rc3                       
5.11.0-rc2                       
5.10.130                         
5.10.125                         
5.10.120                         
5.10.115                         
5.10.110                         
5.10.105                         
5.10.100                         
5.10.95                          
5.10.90                          
5.10.85                          
5.10.80                          
5.10.75                          
5.10.70                          
5.10.65                          
5.10.60                          
5.10.55                          
5.10.50                          
5.10.35                          
5.10.30                          
5.10.25                          
5.10.20                          
5.10.15                          
5.10.10                          
5.10.5                           
5.10.0                           
5.10.0-rc6                       
5.10.0-rc5                       
5.10.0-rc4                       
5.10.0-rc3                       
5.10.0-rc2                       
5.10.0-rc1                       
5.9.16                           
5.9.15                           
5.9.12                           
5.9.10                           
5.9.8                            
5.9.7                            
5.9.0                            
5.9.0-rc8                        
5.9.0-rc7                        
5.9.0-rc6                        
5.9.0-rc5                        
5.9.0-rc4                        
5.9.0-rc3                        
5.9.0-rc2                        
5.9.0-rc1                        
5.8.18                           
5.8.17                           
5.8.16                           
5.8.15                           
5.8.14                           
5.8.13                           
5.8.12                           
5.8.11                           
5.8.10                           
5.8.9                            
5.8.5                            
5.8.0                            
5.8.0-rc7                        
5.8.0-rc6                        
5.8.0-rc5                        
5.8.0-rc4                        
5.8.0-rc3                        
5.8.0-rc2                        
5.8.0-rc1                        
5.7.19                           
5.7.18                           
5.7.17                           
5.7.16                           
5.7.15                           
5.7.14                           
5.7.13                           
5.7.12                           
5.7.11                           
5.7.10                           
5.7.5                            
5.7.0                            
5.7.0-rc7                        
5.7.0-rc6                        
5.7.0-rc5                        
5.7.0-rc4                        
5.7.0-rc3                        
5.7.0-rc2                        
5.7.0-rc1                        
5.6.19                           
5.6.18                           
5.6.17                           
5.6.16                           
5.6.15                           
5.6.14                           
5.6.13                           
5.6.11                           
5.6.10                           
5.6.9                            
5.6.5                            
5.6.0                            
5.6.0-rc7                        
5.6.0-rc6                        
5.6.0-rc5                        
5.6.0-rc4                        
5.6.0-rc3                        
5.6.0-rc2                        
5.6.0-rc1                        
5.5.19                           
5.5.18                           
5.5.17                           
5.5.16                           
5.5.15                           
5.5.14                           
5.5.13                           
5.5.12                           
5.5.11                           
5.5.10                           
5.5.5                            
5.5.0                            
5.5.0-rc7                        
5.5.0-rc6                        
5.5.0-rc5                        
5.5.0-rc4                        
5.5.0-rc3                        
5.5.0-rc2                        
5.5.0-rc1                        
5.4.228                          
5.4.227                          
5.4.226                          
5.4.224                          
5.4.223                          
5.4.222                          
5.4.221                          
5.4.220                          
5.4.219                          
5.4.218                          
5.4.217                          
5.4.216                          
5.4.215                          
5.4.214                          
5.4.205                          
5.4.200                          
5.4.190                          
5.4.185                          
5.4.180                          
5.4.175                          
5.4.170                          
5.4.165                          
5.4.160                          
5.4.155                          
5.4.150                          
5.4.145                          
5.4.140                          
5.4.135                          
5.4.130                          
5.4.125                          
5.4.120                          
5.4.115                          
5.4.110                          
5.4.105                          
5.4.100                          
5.4.95                           
5.4.90                           
5.4.85                           
5.4.75                           
5.4.70                           
5.4.65                           
5.4.60                           
5.4.55                           
5.4.50                           
5.4.45                           
5.4.25                           
5.4.20                           
5.4.15                           
5.4.10                           
5.4.5                            
5.4.0                            
5.4.0-rc8                        
5.4.0-rc7                        
5.4.0-rc6                        
5.4.0-rc5                        
5.4.0-rc4                        
5.4.0-rc3                        
5.4.0-rc2                        
5.4.0-rc1                        
5.3.18                           
5.3.17                           
5.3.16                           
5.3.15                           
5.3.14                           
5.3.13                           
5.3.12                           
5.3.11                           
5.3.10                           
5.3.9                            
5.3.5                            
5.3.0                            
5.3.0-rc8                        
5.3.0-rc7                        
5.3.0-rc6                        
5.3.0-rc5                        
5.3.0-rc4                        
5.3.0-rc3                        
5.3.0-rc2                        
5.3.0-rc1                        
5.2.21                           
5.2.20                           
5.2.19                           
5.2.18                           
5.2.17                           
5.2.16                           
5.2.15                           
5.2.14                           
5.2.13                           
5.2.12                           
5.2.10                           
5.2.5                            
5.2.0                            
5.2.0-rc7                        
5.2.0-rc6                        
5.2.0-rc5                        
5.2.0-rc4                        
5.2.0-rc3                        
5.2.0-rc2                        
5.2.0-rc1                        
5.1.21                           
5.1.20                           
5.1.19                           
5.1.18                           
5.1.17                           
5.1.16                           
5.1.15                           
5.1.14                           
5.1.13                           
5.1.12                           
5.1.10                           
5.1.5                            
5.1.0                            
5.1.0-rc7                        
5.1.0-rc6                        
5.1.0-rc5                        
5.1.0-rc4                        
5.1.0-rc3                        
5.1.0-rc2                        
5.1.0-rc1                        
5.0.21                           
5.0.20                           
5.0.19                           
5.0.18                           
5.0.17                           
5.0.16                           
5.0.15                           
5.0.14                           
5.0.13                           
5.0.12                           
5.0.10                           
5.0.5                            
5.0.0                            
5.0.0-rc8                        
5.0.0-rc7                        
5.0.0-rc6                        
5.0.0-rc5                        
5.0.0-rc4                        
5.0.0-rc3                        
5.0.0-rc2                        
5.0.0-rc1    
```

**Comment by Weuxel** [comment]
Same.

**Comment by korikori** [comment]
I experienced the same, but kind of resolved it by clearing the cache (`--clean-cache`) and afterwards I was able to update to 6.1.1. I also now see the 6.1.4 kernel (but couldn't install it), and I also don't see any of the 6.2 rc's.

**Comment by korikori** [comment]
I now see why none of the 6.2 rcs are not available - the amd64 builds failed for each version, so there are no packages available. See https://kernel.ubuntu.com/~kernel-ppa/mainline/v6.2-rc1/ as an example.

**Comment by Weuxel** [comment]
Cleaning the cache indeed resolved it for me. Thanks!

**Comment by johnnynunez** [comment]
6.2 versions now are available and builded

**Comment by KaMyKaSii** [comment]
> 6.2 versions now are available and builded https://kernel.ubuntu.com/~kernel-ppa/mainline/v6.2-rc3/ ![image](https://user-images.githubusercontent.com/22727137/211532670-efdedfca-0456-49aa-9001-7c53198dd990.png)

Kernel 6.2-rc3 does not boot Linux Mint  21.1, grub returns error "kernel doesn't support EFI handover". Does it work for you?

**Comment by johnnynunez** [comment]
> > 6.2 versions now are available and builded https://kernel.ubuntu.com/~kernel-ppa/mainline/v6.2-rc3/ ![image](https://user-images.githubusercontent.com/22727137/211532670-efdedfca-0456-49aa-9001-7c53198dd990.png)
> 
> Kernel 6.2-rc3 does not boot Linux Mint 21.1, grub returns error "kernel doesn't support EFI handover". Does it work for you?

the same problem, this problem is because they don't compile with this option ON

**Comment by korikori** [comment]
Yes, it's a problem in how this particular image was built (as well as a bug in grub2, see https://bugs.launchpad.net/ubuntu/+source/grub2/+bug/1895912), but in any case it has little to do with the `mainline` app and I believe the original subject of this issue has been resolved.

**Comment by bkw777** [comment]
Failed and incomplete builds are not included.

RC versions are not included unless you enable RC builds in settings.

Until just a day or so ago there was a long period of nothing but failed builds, and so nothing to show after 6.0.9 until just now.

Booting/running problems are not this app's responsibility, only downloading and installing/removing packages provided by the mainline-ppa site. *every* mainline kernel is always to be treated as broken by default, per the mainline-ppa site itself and ubuntu wiki. Usually they mostly work fine, and that is just luck. When one doesn't work, all it means is don't use that one. This app can not predict, prevent, track, warn, or do anything at all about that. If you need more safety, do not run mainline kernels.

**Comment by TeoColuccio** [comment]
> Failed and incomplete builds are not included.

That's not the problem.
I think there is a problem with the cache because even now, from the GUI and having the RCs enabled, the latest versions I can download are 6.1.4 and 6.1.0-rc5 for the RCs. 
From the terminal, after `mainline --clean-cache`, making me print the list of available ones, also 6.2.0-rc3 came out. And I repeat, I had already had the RCs enabled for a long time, in fact I could already see them as soon as the issue was opened.

**Comment by bkw777** [comment]
Reopening until I can look at it. Cache problems and problems parsing the web site are definitely things with this app.
